### PR TITLE
Fix color contrast in code editor.

### LINKED ISSUE #25562: "Code editor" mode: fix borders and placeholders color contrast
**Describe the bug**

As noted in the similar issue (#25539) for the "Edit as HTML" styles, also the "Code editor" mode isn't meant to be WYSIWYG thus there's no reason why the input fields (actually textareas) used in this mode should not meet accessibility guidelines for contrast color.

Worth reminding in "Code editor" mode both the title and the post content use native HTML textarea elements.

**1. The borders**
- Normal state: the textarea elements border color is `#ccc` which against a white background produces a contrast ratio of 1.61:1 while it should be at least 3:1 (non-text contrast).

Screenshot 2020-09-23 at 15 35 35

- Focus state: the border color is `#1e1e1e` which is OK but since there's the need to make the border normal state color darker, this would likely need to be darkened as well.
- Focus state: I'd consider to make the focus style color blue as it's the case for all the other native textarea elements and inputs: not sure why it needs to be a dark grey. As said above, the purpose of this mode isn't to provide a WYSIWYG mode: these are native HTML textarea elements.

**2. The native placeholder text**
- For the post title, the color used for the native placeholder text is `rgba(30, 30, 30, 0.55)` which against a white background produces an insufficient contrast ratio of 3.79:1, see https://jdlsn.com/color/?type=rgb&color=30,%2030,%2030&color2=255,255,255&alpha=0.55. Since this is text, the contrast should be at least 4.5:1
- For the post content, the placeholder text color is `#72777c` which is OK but this is the color inherited from the core `forms.css`. I guess the block editor should use its own styling.

Screenshot 2020-09-23 at 15 42 21

Worth noting that core makes sure that all native placeholder text have sufficient color contrast. I'd expect Gutenberg to follow this core best practice. See:

https://github.com/WordPress/wordpress-develop/blob/master/src/wp-admin/css/forms.css#L228-L239

(Firefox also needs the opacity to be reset to 1)

**To reproduce**
- create a new post: do not add any content
- switch fo Code editor mode
- observe the insufficient color contrast of the borders and of the title placeholder

**Editor version (please complete the following information):**
- WordPress version: 5.6 trunk
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? both
- If the Gutenberg plugin is installed, which version is it? 9.0.0

### PR DESCRIPTION
**Pull request by ZebulanStanphill**
## Description
Fixes #25562.

### Before
image
image
image

### After
image
image
image

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by aristath — APPROVED** [review]

File: packages/edit-post/src/components/text-editor/style.scss
Comment (jasmussen) [inline packages/edit-post/src/components/text-editor/style.scss:21]: The border doesn't need to meet more than 3:1 ratio (UI and large text), and by having a lighter gray it provides contrast between neutral and focused state.

Would appreciate that changed to $gray-600 before merging.

Comment (afercia) [reply packages/edit-post/src/components/text-editor/style.scss:21]: Agreed the borders need only a 3:1 contrast. Also mentioned in the issue:
>  it should be at least 3:1 (non-text contrast).

The placeholder text still requires a 4.5:1 contrast ratio.

Comment (ZebulanStanphill) [reply packages/edit-post/src/components/text-editor/style.scss]: Alright, I've changed the unselected border colors to `$gray-600` and will merge when tests pass.

image

**Pull request merged**
